### PR TITLE
Rename gcloud-java to google-cloud

### PR DESCRIPTION
**Pull request by mziccard**
This PR renames `gcloud-java` to `google-cloud` and applies the necessary changes to docs.

Changes to codebase required for a successful renaming:
- [x] Update `pom.xml` files and rename directories
- [x] Update links to files in javadoc
- [x] Add `GOOGLE_CLOUD_PROJECT` as default variable for the project id
- [x] Bump to new snapshot version `0.3.0-SNAPSHOT` 
- [x] Rename `gcloud-java` to `google-cloud` in landing page for future releases
- [x] Update build scripts to work on new `google-cloud-*` directory structure

Requirements before merging:
- [x] Rename github project from `gcloud-java` to `google-cloud-java`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,26 +17,26 @@ Maven 3.0+ is required.
 
 When downloading the source, we recommend you obtain service account credentials. 
 These credentials will allow you to run integration tests using `mvn verify` in command line. 
-Follow step 2 of the [authentication instructions](https://github.com/GoogleCloudPlatform/gcloud-java#authentication) to generate and use JSON service account credentials.
+Follow step 2 of the [authentication instructions](https://github.com/GoogleCloudPlatform/google-cloud-java#authentication) to generate and use JSON service account credentials.
 
 It's also important to test that changes don't break compatibility with App/Compute Engine and when running elsewhere. 
-To run tests on different platforms, try deploying the apps available on the [gcloud-java-examples](https://github.com/GoogleCloudPlatform/gcloud-java-examples) repository. 
-End-to-end tests should ensure that gcloud-java works when running on the
+To run tests on different platforms, try deploying the apps available on the [google-cloud-examples](https://github.com/GoogleCloudPlatform/google-cloud-examples) repository.
```
Comment (garrettjonesgoogle) [inline CONTRIBUTING.md]: Won't this need to be google-cloud-java-examples?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -17,26 +17,26 @@ Maven 3.0+ is required.
...
+End-to-end tests should ensure that google-cloud works when running on the
```
Comment (garrettjonesgoogle) [inline CONTRIBUTING.md]: google-cloud-java?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -17,26 +17,26 @@ Maven 3.0+ is required.
...
+When changes are made to authentication and project ID-related code, authentication and project ID inference should be tested using all relevant methods detailed in the [authentication docs](https://github.com/GoogleCloudPlatform/google-cloud-java#authentication) and [project ID docs](https://github.com/GoogleCloudPlatform/google-cloud-java#specifying-a-project-id).
 
 Known issue: If you have installed the Google Cloud SDK, be sure to log in (using `gcloud auth login`) before running tests. Though the Datastore tests use a local Datastore emulator that doesn't require authentication, they will not run if you have the Google Cloud SDK installed but aren't authenticated.
 
 **Please, do not use your production projects for executing integration tests.** While we do our best to make our tests independent of your project's state and content, they do perform create, modify and deletes, and you do not want to have your production data accidentally modified.
 
 Adding Features
 ---------------
-In order to add a feature to gcloud-java:
+In order to add a feature to google-cloud:
```
Comment (garrettjonesgoogle) [inline CONTRIBUTING.md]: google-cloud-java?

File: README.md [hunk 4]
```diff
@@ -27,66 +27,69 @@ This client supports the following Google Cloud Platform services:
 > Note: This client is a work-in-progress, and may occasionally
 > make backwards-incompatible changes.
 
+## Where did `gcloud-java` go?
+
+`gcloud-java` lives on under a new name, `google-cloud`.
```
Comment (garrettjonesgoogle) [inline README.md]: google-cloud-java?

Comment (garrettjonesgoogle) [inline README.md]: Or not so simple: The repo is `google-cloud-java` and the artifacts are `google-cloud`.

File: CONTRIBUTING.md (hunk 1 above)
Comment (mziccard) [inline CONTRIBUTING.md]: Actually we should rollback this change until we update the other repo.

File: CONTRIBUTING.md (hunk 2 above)
Comment (mziccard) [inline CONTRIBUTING.md]: Not sure whether we should use the artifact or the repo name here. Probably the repo name.

File: README.md (hunk 4 above)
Comment (mziccard) [inline README.md]: This is in line with what Node guys did, I would keep as is.

**Comment by mziccard** [comment]
After some struggle all Veneer Toolkit PRs are merged in and renaming PR is ready. If no major issues arise I will rename the repo and merge this in first thing tomorrow.

/cc @lesv @jgeewax @omaray

**Comment by lesv** [comment]
SGTM - do we have someone engaged for any documentation changes?

**Comment by mziccard** [comment]
> do we have someone engaged for any documentation changes?

Not that I am aware of. GoogleCloudPlatform site was updated to handle redirects on 404 such as:
```
http://googlecloudplatform.github.io/gcloud-* -> http://googlecloudplatform.github.io/google-cloud-*
```
So we should not suffer broken links on that. Links to code on github will still be broken due to directory renaming. Sadly, I have no control on that.

**Comment by lesv** [comment]
I'll see what I can do on this.

**Comment by lesv** [comment]
When will you update Maven Central?

**Comment by mziccard** [comment]
> When will you update Maven Central?

After this gets merged I'll release `google-cloud 0.3.0`.

**Comment by lesv** [comment]
Are you also regenerating the googlecloudplatform.github.io ??

**Comment by lesv** [comment]
Ah, just found what I was looking for.

**Comment by omaray** [comment]
Great job everyone!! And thanks Les for your help with c.g.c!

**Pull request merged**